extensions/sql-migration/src/wizard/skuRecommendationPage.ts

```diff
@@ -254,49 +198,74 @@ export class SKURecommendationPage extends MigrationWizardPage {
 				descriptions
 			});
 		});
-		let miDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, 'mi');
-		let vmDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, 'vm');
+		let miDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, MigrationTargetType.SQLMI);
+		let vmDialog = new AssessmentResultsDialog('ownerUri', this.migrationStateModel, 'Assessment Dialog', this, MigrationTargetType.SQLVM);
 
 		this._rbg.onLinkClick(async (value) => {
-
-			//check which card is being selected, and open correct dialog based on link
-			if (value.description.linkDisplayValue === 'View/Change') {
-				if (value.cardId === 'AzureSQLVM') {
+			if (value.cardId === MigrationTargetType.SQLVM) {
+				this._rbg.selectedCardId = MigrationTargetType.SQLVM;
+				if (value.description.linkDisplayValue === 'View/Change') {
 					await vmDialog.openDialog();
-				} else if (value.cardId === 'AzureSQLMI') {
-					await miDialog.openDialog();
-				}
-			} else if (value.description.linkDisplayValue === 'Learn more') {
-				if (value.cardId === 'AzureSQLVM') {
+				} else if (value.description.linkDisplayValue === 'Learn more') {
 					vscode.env.openExternal(vscode.Uri.parse('https://docs.microsoft.com/en-us/azure/azure-sql/virtual-machines/windows/sql-server-on-azure-vm-iaas-what-is-overview'));
-				} else if (value.cardId === 'AzureSQLMI') {
+				}
+			} else if (value.cardId === MigrationTargetType.SQLMI) {
+				this._rbg.selectedCardId = MigrationTargetType.SQLMI;
+				if (value.description.linkDisplayValue === 'View/Change') {
+					await miDialog.openDialog();
+				} else if (value.description.linkDisplayValue === 'Learn more') {
 					vscode.env.openExternal(vscode.Uri.parse('https://docs.microsoft.com/en-us/azure/azure-sql/managed-instance/sql-managed-instance-paas-overview '));
 				}
 			}
 		});
 
 		this._rbg.onSelectionChanged((value) => {
-			this.populateResourceInstanceDropdown();
+			this.toggleTargetTypes(value.cardId);
 		});
 
-		this._rbg.selectedCardId = 'AzureSQLMI';
+		this._rbg.selectedCardId = MigrationTargetType.SQLMI;
+
+
+		const component = view.modelBuilder.divContainer().withItems(
+			[
+				this._rbg
+			]
+		).component();
+		return component;
+	}
+
+	private toggleTargetTypes(targetType: string) {
```

Comment:
This isn't toggling anything - it appears to be just updating the UI based on the current targetType. Would suggest finding a better name for the function since toggle has a very specific meaning.